File: server/apps/nexotype/schemas/commercial/patent_assignee_schemas.py
from pydantic import BaseModel, ConfigDict
from datetime import date, datetime

# ==========================================
# Request Schemas
# ==========================================

class PatentAssigneeCreate(BaseModel):
    """Schema for creating a new patent assignee"""
    patent_id: int
    market_organization_id: int
    assignment_date: date


class PatentAssigneeUpdate(BaseModel):
    """Schema for updating a patent assignee"""
    patent_id: int = None
    market_organization_id: int = None
    assignment_date: date = None


# ==========================================
# Response Schemas
# ==========================================

class PatentAssigneeDetail(BaseModel):
    """Schema for patent assignee details"""
    id: int
    patent_id: int
    market_organization_id: int
    assignment_date: date
    created_at: datetime
    updated_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class PatentAssigneeListResponse(BaseModel):
    """Response schema for listing patent assignees"""
    success: bool
    data: list[PatentAssigneeDetail] | None = None
    count: int = 0
    error: str | None = None


class PatentAssigneeResponse(BaseModel):
    """Response schema for single patent assignee operations"""
    success: bool
    data: PatentAssigneeDetail | None = None
    error: str | None = None


class MessageResponse(BaseModel):
    """Simple message response"""
    success: bool
    message: str | None = None
    error: str | None = None

File: client/src/modules/nexotype/schemas/commercial/patent-assignee.schemas.ts
import { z } from 'zod';

// ==========================================
// Full Schema (GET responses)
// ==========================================

export const PatentAssigneeSchema = z.object({
  id: z.number(),
  patent_id: z.number().int(),
  market_organization_id: z.number().int(),
  assignment_date: z.string(),
  created_at: z.string(),
  updated_at: z.string().nullable(),
});

// ==========================================
// Input Schemas (POST/PUT)
// ==========================================

export const CreatePatentAssigneeSchema = z.object({
  patent_id: z.number().int(),
  market_organization_id: z.number().int(),
  assignment_date: z.string().min(1),
});

export const UpdatePatentAssigneeSchema = z.object({
  patent_id: z.number().int().optional(),
  market_organization_id: z.number().int().optional(),
  assignment_date: z.string().min(1).optional(),
});

// ==========================================
// Types
// ==========================================

export type PatentAssignee = z.infer<typeof PatentAssigneeSchema>;
export type CreatePatentAssignee = z.infer<typeof CreatePatentAssigneeSchema>;
export type UpdatePatentAssignee = z.infer<typeof UpdatePatentAssigneeSchema>;

// ==========================================
// Response Types
// ==========================================

export type PatentAssigneeResponse = {
  success: boolean;
  data?: PatentAssignee;
  error?: string;
};

export type PatentAssigneesResponse = {
  success: boolean;
  data?: PatentAssignee[];
  count?: number;
  error?: string;
};
